sqlglot/optimizer/expand_join_constructs.py

```diff
@@ -0,0 +1,33 @@
+import typing as t
+
+from sqlglot import exp
+from sqlglot.optimizer.scope import traverse_scope
+
+
+def expand_join_constructs(expression: exp.Expression) -> exp.Expression:
+    """
+    Replace "join constructs" (*) by equivalent SELECT * subqueries.
+
+    Example:
+    >>> import sqlglot
+    >>> expression = sqlglot.parse_one("SELECT * FROM (tbl1 AS tbl1 JOIN tbl2 AS tbl2 ON id1 = id2) AS tbl")
+    >>> expand_join_constructs(expression).sql()
+    'SELECT * FROM (SELECT * FROM tbl1 AS tbl1 JOIN tbl2 AS tbl2 ON id1 = id2) AS tbl'
+
+    (*) See section 7.2.1.2 in https://www.postgresql.org/docs/current/queries-table-expressions.html
+    """
+    for scope in traverse_scope(expression):
```

Comment:
Do you need scope here? I.e. what does it add beyond expression.walk?

Reply:
Hmm, that's a good question. It seemed like a natural choice to me at first because it captures the `(tbl as tbl)` expressions nicely (i.e. it captures the subqueries as scopes and we can just check if the expression is an `exp.Table`).

But now that you mention it, we can probably walk the tree and look for `exp.Subquery` expressions for which the `.this.unnest()` returns an `exp.Table` and do the same thing? Does that make sense?